test/Microsoft.AspNetCore.Razor.Runtime.Test/TagHelpers/ReadOnlyTagHelperAttributeListTest.cs

```diff
@@ -668,33 +669,34 @@ public void ModifyingUnderlyingAttributes_AffectsExposedAttributes()
         [Theory]
         [MemberData(nameof(StringIndexerData))]
         public void ModifyingUnderlyingAttributes_StringIndexer_ReturnsExpectedResult(
-            IEnumerable<IReadOnlyTagHelperAttribute> initialAttributes,
+            IEnumerable<TagHelperAttribute> initialAttributes,
             string nameToLookup,
-            IReadOnlyTagHelperAttribute expectedAttribute)
+            TagHelperAttribute expectedAttribute)
         {
             // Arrange
-            var attributes = new TestableReadOnlyTagHelperAttributes(Enumerable.Empty<IReadOnlyTagHelperAttribute>());
+            var attributes = new TestableReadOnlyTagHelperAttributes(Enumerable.Empty<TagHelperAttribute>());
             attributes.PublicAttributes.AddRange(initialAttributes);
+            TagHelperAttribute attribute;
 
             // Act
-            var attribute = attributes[nameToLookup];
+            attributes.TryGetAttribute(nameToLookup, out attribute);
```

Comment:
See a few more changes away from the indexer later in this PR. Revert all of that because neither the indexer nor `TryGetAttribute()` are new or significantly changed. If `TryGetAttribute()` isn't tested sufficiently, add tests for it.